features/product/managing_product_variants/seeing_correct_catalog_price_history_after_creating_product_variant.feature

```diff
@@ -0,0 +1,60 @@
+    In order to be aware of historical product prices
+    As an Administrator
+    I want to see the catalog price history of the product I've created
+
+    Background:
+        Given the store operates on a single channel in "United States"
+        And I am logged in as an administrator
+
+    @todo
+    Scenario: Seeing historical product variant prices after the product variant has been created without any promotion applied
+        When I want to create a new simple product
+        And I specify its code as "WYBOROWA_VODKA"
+        And I name it "Wyborowa Vodka" in "English (United States)"
+        And I set its slug to "wyborowa-vodka" in "English (United States)"
+        And I set its price to "$20.00" for "United States" channel
+        And I set its original price to "$25.00" for "United States" channel
+        And I add it
+        And I go to the product price history
+        Then I should see a single log entry in the catalog price history for the "Wyborowa Vodka" variant
+        And there should be a log entry with the "$20.00" selling price, "$25.00" original price and datetime of the price change
+
+    @todo
+    Scenario: Seeing historical product variant prices after the product variant has been created without original price and any promotion applied
+        When I want to create a new simple product
+        And I specify its code as "WYBOROWA_VODKA"
+        And I name it "Wyborowa Vodka" in "English (United States)"
+        And I set its slug to "wyborowa-vodka" in "English (United States)"
+        And I set its price to "$20.00" for "United States" channel
+        And I add it
+        And I go to the product price history
+        Then I should see a single log entry in the catalog price history for the "Wyborowa Vodka" variant
+        And there should be a log entry with the "$20.00" selling price, "$20.00" original price and datetime of the price change
+
+    @todo
+    Scenario: Seeing historical product variant prices after the product variant has been created without catalog price and any promotion applied
+        When I want to create a new simple product
+        And I specify its code as "WYBOROWA_VODKA"
+        And I name it "Wyborowa Vodka" in "English (United States)"
+        And I set its slug to "wyborowa-vodka" in "English (United States)"
+        And I set its original price to "$25.00" for "United States" channel
+        And I add it
+        And I go to the product price history
+        Then I should see a single log entry in the catalog price history for the "Wyborowa Vodka" variant
+        And there should be a log entry with the "$25.00" selling price, "$25.00" original price and datetime of the price change
+
+    @todo
+    Scenario: Seeing historical product variant prices after the product variant has been created with catalog promotions applied
+        Given there is a catalog promotion "Christmas sale" that reduces price by "50%" and applies on "Wyborowa Vodka" product
```

Comment:
I wonder if will this scenario can be realized since the promotion is applied before the product is created 🤔

Reply:
I'm fairly sure we should take this scenario into account because even if it's applied at the same time product is created, the CP works async so the promotion will be applied in a delay.

Reply:
Ok I see now what is a problem here – will fix that